### DISCUSSION TITLE
"Standard" style in "odt_styles_file" is not used, it has no effect at all

### DISCUSSION THREAD

**Discussion by QiangF** [kjambunathan/org-mode-ox-odt#127] [Q&A] [answered]
I mean this one:

    <style:style style:name="Standard" style:family="paragraph" style:class="text">

But setting it with odt_extra_styles works:

    #+odt_extra_styles: <style:style style:name="Standard" style:family="paragraph">
    #+odt_extra_styles: <style:paragraph-properties fo:line-height="300%"
    #+odt_extra_styles:                             fo:padding="0cm"
    #+odt_extra_styles:                             fo:border="none">
    #+odt_extra_styles: </style:paragraph-properties>
    #+odt_extra_styles: </style:style>

To make the former work is more preferable, because it looks very bad with all these extra lines.

**Comment by kjambunathan** [comment]
> <style:style style:name="Standard" style:family="paragraph" style:class="text">

The paragraph style of "Standard" comes from this in `OrgOdtStyles.xml`.

```xml
<style:default-style style:family="paragraph">
  <style:paragraph-properties fo:hyphenation-ladder-count="no-limit"
			      style:text-autospace="ideograph-alpha"
			      style:punctuation-wrap="hanging" style:line-break="strict"
			      style:tab-stop-distance="1.251cm" style:writing-mode="page" />
  <style:text-properties style:use-window-font-color="true"
			 style:font-name="Times New Roman" fo:font-size="12pt"
			 fo:language="en" fo:country="GB" style:letter-kerning="true"
			 style:font-name-asian="SimSun" style:font-size-asian="12pt"
			 style:language-asian="zh" style:country-asian="CN"
			 style:font-name-complex="Tahoma" style:font-size-complex="12pt"
			 style:language-complex="hi" style:country-complex="IN"
			 fo:hyphenate="false" fo:hyphenation-remain-char-count="2"
			 fo:hyphenation-push-char-count="2" />
</style:default-style>
```

**Comment by QiangF** [comment]
The user's "Standard" style in the style file is shadowed by "Standard" style appended in function org-odt-write-styles-file.

**Comment by QiangF** [comment]
I want the following two styles different: 
1. Default: <style:default-style style:family="paragraph">
2. Standard: <style:style style:name="Standard" style:family="paragraph" style:class="text">

**Comment by kjambunathan** [comment]
I am having trouble understanding what you are saying ... If I give some more time to think about it, I will be able to figure out your needs.  Did the recent changes break any of the new functionality? 

Btw, "Gridded Table" is based on "CustomContents".  It is in `OrgOdtContentTemplate.xml". It derives from "Table_20_Contents" which is undefined.  And I will move the "Custom" paragraph styles from content tempate to styles file very soon.

**Comment by QiangF** [comment]
No, I mean this function "org-odt-write-styles-file" overwrite user's "Standard" style in OrgOdtStyles.xml.
This is not related to my other posts. Sorry it confuses you.

**Reply by kjambunathan** [reply]
> No, I mean this function "org-odt-write-styles-file" overwrite user's "Standard" style in OrgOdtStyles.xml.
> This is not related to my other posts. Sorry it confuses you.

> No, I mean this function "org-odt-write-styles-file" overwrite user's "Standard" style in OrgOdtStyles.xml.
> This is not related to my other posts. Sorry it confuses you.

Can you point me to where it does so?  

May be it is the locale related changes that is creating the problem ...

Are you using the "language" feature in experimental?  May be look at this diagram https://github.com/kjambunathan/org-mode-ox-odt/blob/803ae54b30b79bc86d838a94bc8e26c058f6f390/lisp/ox-odt.el#L2820 and see if it makes any sense.  Also look at 'OrgUser` style mentioned in https://github.com/kjambunathan/org-mode-ox-odt/issues/80#issuecomment-943206601

**Reply by kjambunathan** [reply]
Ok I think I understand what you are saying ....

This is the place I install "Standard" and it inherits from "OrgUser".  

https://github.com/kjambunathan/org-mode-ox-odt/blob/803ae54b30b79bc86d838a94bc8e26c058f6f390/lisp/ox-odt.el#L2860

OrgUser will be the "default-style" and "Standard" inherits from "OrgUser" and gets the "default-style.  And in your case, "Standard" is DIFFERENT from "default-style".

I think you need to rename the "Standard" to "OrgUser".  Your problem will be solved.

**_(I thought you are complaining about paragraph in tables.  But you are talking about changes introduced by "locale" feature)_**

**Comment by QiangF** [comment]
Here we may need to parse the xml to lisp object first, and merge the change, instead of brutally overwrite.

**Comment by QiangF** [comment, marked as answer]
I am not sure. I don't enable the language experimetal feature.
At least my change to "Standard" in OrgOdtStyles.xml has no effect.

    <style:style style:name="Standard" style:family="paragraph" style:class="text">
    <loext:graphic-properties draw:fill="none" draw:fill-color="#729fcf"/>
    <style:paragraph-properties fo:margin-left="0cm"
                                fo:margin-right="0cm"
                                fo:margin-top="0cm"
                                fo:margin-bottom="0cm"
                                fo:line-height="150%"
                                fo:text-indent="0.0cm"
                                loext:contextual-spacing="false"
                                style:auto-text-indent="false"
                                style:page-number="auto"
                                fo:background-color="transparent"
                                style:shadow="none">
        <style:tab-stops/>
    </style:paragraph-properties>
    </style:style>

**Reply by kjambunathan** [reply]
> At least my change to "Standard" in OrgOdtStyles.xml has no effect.

See my other reply.  Replace "Standard" with "OrgUser"

**Comment by QiangF** [comment]
It works. Thank you so much!
